__integrations__/index.test.ts

```diff
@@ -505,6 +508,78 @@ describe("Create Advanced Message", () => {
     }
   );
 
+  it("should return the message without Read status, if user is allowed to read it", async () => {
+    const fiscalCode = anAutoFiscalCode;
+    const serviceId = aValidServiceId;
+
+    const body = {
+      message: {
+        fiscal_code: fiscalCode,
+        feature_level_type: "ADVANCED",
+        content: aMessageContent
+      }
+    };
+
+    const nodeFetchWithoutPermission = getNodeFetch({
+      "x-subscription-id": serviceId
+    });
+    const nodeFetch = getNodeFetch({
+      "x-subscription-id": serviceId,
+      "x-user-groups":
+        customHeaders["x-user-groups"] +
+        ",ApiMessageWriteAdvanced,ApiMessageReadAdvanced"
+    });
+
+    const result = await postCreateMessage(nodeFetch)(body);
+
+    expect(result.status).toEqual(201);
+
+    const messageId = ((await result.json()) as CreatedMessage).id;
+    expect(messageId).not.toBeUndefined();
+
+    // Wait the process to complete
+    await delay(WAIT_MS);
+
+    // Check response having `ApiMessageReadAdvanced` authorization
+    const resultGet = await getSentMessage(nodeFetch)(fiscalCode, messageId!);
+
+    expect(resultGet.status).toEqual(200);
+    const detail = (await resultGet.json()) as ExternalMessageResponseWithContent;
+
+    expect(detail).toMatchObject(
+      expect.objectContaining({
+        message: expect.objectContaining({
+          id: messageId,
+          ...body.message
+        }),
+        status: MessageStatusValueEnum.PROCESSED,
+        read_status: ReadStatusEnum.UNAVAILABLE
+      })
+    );
+
+    // Check response without having `ApiMessageReadAdvanced` authorization
+
+    const resultGetWithoutPermission = await getSentMessage(
+      nodeFetchWithoutPermission
+    )(fiscalCode, messageId!);
+
+    expect(resultGetWithoutPermission.status).toEqual(200);
+    const detailWithoutPermission = (await resultGetWithoutPermission.json()) as ExternalMessageResponseWithContent;
+
+    expect(detailWithoutPermission).toMatchObject(
+      expect.objectContaining({
+        message: expect.objectContaining({
+          id: messageId,
```

Comment:
ditto

Reply:
Here we are checking the response in case service does not own `ApiMessageReadAdvanced`, so we do not send payment status and read status at all.

Reply:
Yes is right, I was referring on `id` overriding
```
{
  id: messageId,
  ...other
}
// vs
{
  ...other,
  id: messageId
}
```

Reply:
Now I get it, sorry... please notice that `id` is never defined in `body.message`, so result is the same.
Anyway I switched it in [a6e9bee](https://github.com/pagopa/io-functions-services/pull/211/commits/a6e9bee01d0259c7c7970a42ac39cc59bcadae1c) for readability, thanks.